File: src/main/java/com/ject/studytrip/member/domain/policy/MemberPolicy.java
package com.ject.studytrip.member.domain.policy;

import com.ject.studytrip.global.exception.CustomException;
import com.ject.studytrip.member.domain.error.MemberErrorCode;
import com.ject.studytrip.member.domain.model.Member;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class MemberPolicy {
    public static void validateNotDuplicated(boolean exists) {
        if (exists) {
            throw new CustomException(MemberErrorCode.MEMBER_ALREADY_EXISTS);
        }
    }

    public static void validateNotDeleted(Member member) {
        if (member.getDeletedAt() != null) {
            throw new CustomException(MemberErrorCode.MEMBER_ALREADY_DELETED);
        }
    }

    public static void validateDeleted(Member member) {
        if (member.getDeletedAt() == null) {
            throw new CustomException(MemberErrorCode.MEMBER_NOT_DELETED);
        }
    }
}

File: src/main/kotlin/com/ject/studytrip/member/domain/policy/MemberPolicy.kt
package com.ject.studytrip.member.domain.policy

import com.ject.studytrip.global.exception.CustomException
import com.ject.studytrip.member.domain.error.MemberErrorCode
import com.ject.studytrip.member.domain.model.Member

object MemberPolicy {
    fun validateNotDeleted(member: Member) {
        if (member.isDeleted()) {
            throw CustomException(MemberErrorCode.MEMBER_ALREADY_DELETED)
        }
    }

    fun validateDeleted(member: Member) {
        if (!member.isDeleted()) {
            throw CustomException(MemberErrorCode.MEMBER_NOT_DELETED)
        }
    }

    fun validateNotDuplicated(exists: Boolean) {
        if (exists) {
            throw CustomException(MemberErrorCode.MEMBER_ALREADY_EXISTS)
        }
    }
}
